arangod/StorageEngine/TransactionState.cpp

```diff
@@ -54,7 +56,16 @@ TransactionState::TransactionState(TRI_vocbase_t& vocbase,
       _hints(),
       _options(options),
       _serverRole(ServerState::instance()->getRole()),
-      _registeredTransaction(false) {}
+      _registeredTransaction(false) {
+  if (isCoordinator()) {
+    // for cluser operation we need to set analyzersRevision
+    if (_vocbase.server().hasFeature<arangodb::iresearch::IResearchAnalyzerFeature>()) {
+      _analyzersRevision = _vocbase.server()
+        .getFeature< arangodb::iresearch::IResearchAnalyzerFeature>()
+        .getAnalyzersRevision(_vocbase, true)->getRevision();
+    }
+  }
```

Comment:
instead of lookup feature twice, I'd better do something like:
```
  arangodb::iresearch::IResearchAnalyzerFeature* feature = nullptr;

  try {
    feature = _vocbase.server().getFeature< arangodb::iresearch::IResearchAnalyzerFeature>();
  } catch (...) {
    return;
  }

  _analyzersRevision = feature->getAnalyzersRevision(_vocbase, true)->getRevision();
```